system-tests/lib/crecli/commands.go

```diff
@@ -18,13 +19,28 @@ type CompilationResult struct {
 	ConfigURL   string
 }
 
-func CompileWorkflow(creCLICommandPath, workflowFolder string, configFile *string, settingsFile *os.File) (CompilationResult, error) {
+func CompileWorkflow(creCLICommandPath, workflowFolder string, configFile *string, workflowSettingsFile, settingsFile *os.File) (CompilationResult, error) {
 	var outputBuffer bytes.Buffer
 
 	// the CLI expects the workflow code to be located in the same directory as its `go.mod`` file. That's why we assume that the file, which
 	// contains the entrypoint method is always named `main.go`. This is a limitation of the CLI, which we can't change.
 
-	compileArgs := []string{"workflow", "compile", "-S", settingsFile.Name()}
+	cliFile, err := os.Create(filepath.Join(workflowFolder, CRECLISettingsFileName))
```

Comment:
why? this file is already created, so why do we need to create settings file in the workflow folder?

Reply:
So the compile command gets executed from the other directory where the go file exists. And this step is to copy the cre.yaml file to that directory. Also now CLI needs 2 yaml files, a global `cre.yaml` and then workflow specific `workflow.yaml`